# HG changeset patch
# Date 1510680264 18000
#      Tue Nov 14 12:24:24 2017 -0500
# Node ID 9b87933385e08300ff0389714b795bb70d473446
# Parent  4ce9b333838153df8378764914e7a85c5eefc4bf
Prefix outgoing /me messages with the user name.

diff -r 4ce9b3338381 -r 9b87933385e0 hangouts_conversation.c
--- a/hangouts_conversation.c	Mon Oct 23 22:16:50 2017 +1300
+++ b/hangouts_conversation.c	Tue Nov 14 12:24:24 2017 -0500
@@ -1452,7 +1452,11 @@
 	message_content__init(&message_content);
 	
 	if (purple_message_meify(message_dup, -1)) {
-		//TODO put purple_account_get_private_alias(sa->account) on the front
+		gchar** alias = g_strsplit(purple_account_get_alias(ha->account), " ", -1);
+		gchar* message_prefixed = g_strdup_printf("%s %s", alias[0], message_dup);
+		g_strfreev(alias);
+		g_free(message_dup);
+		message_dup = message_prefixed;
 		
 		event_annotation__init(&event_annotation);
 		event_annotation.has_type = TRUE;
@@ -1478,6 +1482,7 @@
 	
 	g_hash_table_insert(ha->sent_message_ids, g_strdup_printf("%" G_GUINT64_FORMAT, request.event_request_header->client_generated_id), NULL);
 	
+	g_free(message_dup);
 	hangouts_free_segments(segments);
 	hangouts_request_header_free(request.request_header);
 	hangouts_event_request_header_free(request.event_request_header);
diff -r 4ce9b3338381 -r 9b87933385e0 hangouts_events.c
--- a/hangouts_events.c	Mon Oct 23 22:16:50 2017 +1300
+++ b/hangouts_events.c	Tue Nov 14 12:24:24 2017 -0500
@@ -648,8 +648,20 @@
 			EventAnnotation *annotation = chat_message->annotation[i];
 			
 			if (annotation->type == HANGOUTS_MAGIC_HALF_EIGHT_SLASH_ME_TYPE) {
-				//TODO strip name off the front of the first segment
 				purple_xmlnode_insert_data(html, "/me ", -1);
+				// We assume here that the first segment will always be text.
+				// Sanity check in case it's not.
+				if (segments[0]->type == SEGMENT_TYPE__SEGMENT_TYPE_TEXT) {
+					char* text = segments[0]->text;
+					// Skip until we hit whitespace or EOF.
+					for (; *text; ++text) {
+						if (*text == ' ') {
+							// If it's whitespace, advance to the next character and bail.
+							++text; break;
+						}
+					}
+					segments[0]->text = text;
+				}
 				break;
 			}
 		}
